tools/degreeworks-scraper/src/components/PPAPIOfflineClient.ts

```diff
@@ -0,0 +1,34 @@
+import { isErrorResponse } from "@peterportal-api/types";
+import type { Course, RawResponse } from "@peterportal-api/types";
+import fetch from "cross-fetch";
+
+export class PPAPIOfflineClient {
+  private cache = new Map<string, Course>();
+
+  private constructor() {}
+
+  static async new(): Promise<PPAPIOfflineClient> {
+    const ppapi = new PPAPIOfflineClient();
+    const res = await fetch("https://api-next.peterportal.org/v1/rest/courses/all", {
+      headers: { "accept-encoding": "gzip" },
+    });
+    const json: RawResponse<Course[]> = await res.json();
+    if (isErrorResponse(json))
+      throw new Error("Could not fetch courses cache from PeterPortal API");
+    json.payload.forEach((y) => ppapi.cache.set(y.id, y));
+    console.log(
+      `[PPAPIOfflineClient.new] Fetched and stored ${json.payload.length} courses from PeterPortal API`,
+    );
+    return ppapi;
+  }
+
+  getCourse = (courseNumber: string): Course | undefined => this.cache.get(courseNumber);
```

Comment:
I think using an arrow function here is particularly confusing; it's a method that access `this`, which arrow functions are antithetical to.

The main reason I would use arrow function signature is if I need to type the entire function as a whole.

```suggestion
   getCourse (courseNumber: string): Course | undefined {
     return this.cache.get(courseNumber)
   }
```